mixer/pkg/protobuf/yaml/dynamic/builder.go

```diff
@@ -0,0 +1,280 @@
+			descriptor.FieldDescriptorProto_TYPE_SFIXED32,
+			descriptor.FieldDescriptorProto_TYPE_SFIXED64,
+			descriptor.FieldDescriptorProto_TYPE_SINT32,
+			descriptor.FieldDescriptorProto_TYPE_SINT64,
+			descriptor.FieldDescriptorProto_TYPE_ENUM:
+
+			packed := fd.IsPacked() || fd.IsPacked3()
+			if fd.IsRepeated() && !packed {
+				return nil, fmt.Errorf("unpacked primitives not supported: %v", fd)
+			}
+
+			var fld *field
+			if fld, err = c.buildPrimitiveField(v, fd, isMap && k == "key"); err != nil {
+				return nil, fmt.Errorf("unable to encode field: %v. %v", fd, err)
+			}
+			me.fields = append(me.fields, fld)
+		case descriptor.FieldDescriptorProto_TYPE_MESSAGE:
+			m := c.resolver.ResolveMessage(fd.GetTypeName())
+			if m == nil {
+				return nil, fmt.Errorf("unable to resolve message '%s'", fd.GetTypeName())
+			}
+
+			var ma []interface{}
+			if m.GetOptions().GetMapEntry() { // this is a Map
+				ma, err = convertMapToMapentry(v)
+				if err != nil {
+					return nil, fmt.Errorf("unable to process: %v, %v", fd, err)
+				}
+			} else if fd.IsRepeated() { // map entry is always repeated.
+				ma, ok = v.([]interface{})
+				if !ok {
+					return nil, fmt.Errorf("unable to process: %v, got %T, want: []interface{}", fd, v)
+				}
+			} else {
+				ma = []interface{}{v}
+			}
+
+			// now maps, messages and repeated messages all look the same.
+			if err = c.buildMessageField(ma, m, fd, &me); err != nil {
+				return nil, err
+			}
+
+		default:
+			return nil, fmt.Errorf("field not supported '%v'", fd)
+		}
+	}
+
+	// sorting fields is recommended
+	sort.Slice(me.fields, func(i, j int) bool {
+		return me.fields[i].number < me.fields[j].number
+	})
+
+	return me, nil
+}
+
+// buildPrimitiveField handles encoding of single and repeated packed primitives
+func (c Builder) buildPrimitiveField(v interface{}, fd *descriptor.FieldDescriptorProto,
+	static bool) (*field, error) {
+	fld := makeField(fd)
+	e := c.resolver.ResolveEnum(fd.GetTypeName())
+
+	va, packed := v.([]interface{})
+	if !packed {
+		va = []interface{}{v}
+	}
+
+	for _, vl := range va {
+		var err error
+
+		val, isConstString := transFormQuotedString(vl)
+		sval, isString := val.(string)
+
+		var enc Encoder
+
+		// if compiler is nil, everything is considered static
+		if static || isConstString || !isString || c.compiler == nil {
+
+			if fd.IsEnum() {
+				pe := &staticEncoder{name: fd.GetName()}
+				pe.encodedData, err = EncodeEnum(val, pe.encodedData, e.Value)
+				enc = pe
+			} else {
+				enc, err = BuildPrimitiveEncoder(val, fd)
+			}
+
+			if err != nil {
+				return nil, fmt.Errorf("unable to build primitive encoder: %v. %v", val, err)
+			}
+		} else {
+			var expr compiled.Expression
+			var vt v1beta1.ValueType
+
+			if expr, vt, err = c.compiler.Compile(sval); err != nil {
+				return nil, err
+			}
+
+			if fd.IsEnum() {
+				// enums can be expressed as String or INT only
+				if !(vt == v1beta1.INT64 || vt == v1beta1.STRING) {
```

Comment:
we don't need this check in the above static above ?

Reply:
no, `EncodeEnum` handles strings and ints